src/Geolocation/Geolocation.php

```diff
@@ -47,18 +47,81 @@ class Geolocation extends AbstractGeolocation implements GeolocationInterface
 	 */
 	public function register(): void
 	{
-		\add_filter('init', [$this, 'setLocationCookie']);
+		if (!\is_plugin_active('wp-rocket/wp-rocket.php') && !Variables::getGeolocationUseWpRocketAdvancedCache()) {
+			\add_filter('init', [$this, 'setLocationCookie']);
+		}
+
+		// WP Rocket specific hooks.
+		\add_filter('rocket_advanced_cache_file', [$this, 'addNginxAdvanceCacheRules']);
+		\add_filter('rocket_cache_dynamic_cookies', [$this, 'dynamicCookiesList']);
+
 		\add_filter(self::GEOLOCATION_IS_USER_LOCATED, [$this, 'isUserGeolocated'], 10, 3);
 	}
 
+	/**
+	 * List all dynamic cookies that will create new cached version.
+	 *
+	 * @param array<string, mixed> $items Items from the admin.
+	 *
+	 * @return array<int|string, mixed>
+	 */
+	public function dynamicCookiesList(array $items): array
+	{
+		if (!$this->useGeolocation() || !Variables::getGeolocationUseWpRocketAdvancedCache()) {
+			return $items;
+		}
+
+		$items[] = $this->getGeolocationCookieName();
+
+		return $items;
+	}
+
+	/**
+	 * Add geolocation function in advance-cache.php config file on plugin activation used only with Nginx.
+	 *
+	 * @param string $content Original file output.
+	 */
+	public function addNginxAdvanceCacheRules(string $content): string
+	{
+		if (!$this->useGeolocation() || !Variables::getGeolocationUseWpRocketAdvancedCache()) {
+			return $content;
+		}
+
+		$position = \strpos($content, '$rocket_config_class');
+
+		// This part is string on purpose.
+		$output = '
+		$esFormsPath = ABSPATH . "wp-content/plugins/eightshift-forms/src/Geolocation/geolocationDetect.php";
+		if (file_exists($esFormsPath)) {
+			require_once $esFormsPath;
+		};';
```

Comment:
Just a suggestion, but a heredoc might be better here for formatting purposes/readability?

Reply:
I tried, but it just gets printed in the file, and it is not working.